myia/anf_ir.py

```diff
@@ -0,0 +1,304 @@
+"""Intermediate representation definition.
+
+Myia's main intermediate representation (IR) is a graph-based version of ANF.
+Each function definition (lambda) is defined as a graph, consisting of a series
+of function applications.
+
+A function can be applied to a node from another funtion's graph; this
+implicitly creates a nested function. Functions are first-class objects, so
+returning a nested function creates a closure.
+
+"""
+import types
+from ast import AST
+from typing import (List, Set, Tuple, Any, Sequence, MutableSequence,
+                    overload, Iterable)
+
+from myia.ir import Node
+from myia.utils import Named
+
+PARAMETER = Named('PARAMETER')
```

Comment:
Unless this is how we are supposed to check that a node is a parameter, I would replace this by an `UNKNOWN` or `NO_VALUE` token.

Reply:
It's not necessarily how you are supposed to check, but I think it will be easier to understand for people what is going on if `p node.value` in `pdb` prints `PARAMETER` instead of `NO_VALUE`, no?

Reply:
Perhaps. I was thinking that checking if a node is a constant could be done simply with `node.value is not NO_VALUE`, but that's harder if there are multiple tokens that don't represent constants. I suppose a helper function could take care of it anyhow.